walk-in-interview/src/main/webapp/account/firebase-auth.js

```diff
@@ -92,57 +81,66 @@ Auth.addPhoneAuthUI = (elementId, successPath) => {
         // for now, this will only redirect to homepage for exisiting users
-        return !authResult.additionalUserInfo.isNewUser;
+        setCookie(USER_TYPE_COOKIE_PARAM, TYPE_APPLICANT);
+
+        if (authResult.additionalUserInfo.isNewUser) {
+          // Informs user
+          alert(newUserInfo);
+        }
+
+        return true;
       },
+
+      signInFailure: (error) => {
+        console.error(error.message);
+
+        // Back to home page
+        window.location.href = HOMEPAGE_PATH;
+      }
     },
     signInSuccessUrl: successPath,
   });
 };
 
-/**
- * This will create a new applicant account.
- *
- * @param {String} phoneNumber The phone number for the new applicant account.
- * @param {Object} appVerifier The recaptcha verifier.
- */
-Auth.createApplicantAccount = (phoneNumber, appVerifier) => {
-  // TODO(issue/79): set up phone number sign in with otp and recaptcha
-};
-
-/**
- * This will sign into an existing applicant account.
- *
- * @param {String} phoneNumber The number for the existing applicant account.
- * @param {Object} appVerifier The recaptcha verifier.
- */
-Auth.signIntoApplicantAccount = (phoneNumber, appVerifier) => {
-  // TODO(issue/79): set up phone number sign in with otp and recaptcha
-};
-
 /**
  * Signs out the current user.
- *
- * @param {String} elementId The div in which to show the signed out status.
  */
-Auth.signOutCurrentUser = (elementId) => {
+Auth.signOutCurrentUser = () => {
   firebase.auth().signOut().then(() => {
     console.log('sign out successful');
-    document.getElementById(elementId).innerText = STRINGS['sign-out-success'];
+    setCookie(USER_TYPE_COOKIE_PARAM, TYPE_NO_USER);
+    alert(STRINGS['sign-out-success']);
   }).catch((error) => {
     console.error(error);
   });
 };
 
 /**
  * Checks the user sign in status.
+ * Makes a POST request to create a session cookie when the user signs in.
+ * Makes a POST request to clear the session cookie when the user signs out.
  */
-Auth.checkCurrentUser = () => {
+Auth.subscribeToUserAuthenticationChanges = () => {
   firebase.auth().onAuthStateChanged((firebaseUser) => {
-    if (firebaseUser) {
-      console.log('signed in', firebaseUser);
-    } else {
-      console.log('not signed in');
+    if (!firebaseUser) {
+      // User not signed in.
+      console.log('User Not Signed In');
+      // Clears the session cookie
+      return Auth.postIdTokenToSessionLogout(API['log-out']);
     }
+      
+    // User signed in.
+    console.log('User Signed In');
+    // Get the user's ID token as it is needed to exchange
+    // for a session cookie.
+    return firebaseUser.getIdToken()
+        .then((idToken) => {
+          // Session login endpoint is queried and session cookie is set.
+          // CSRF protection should be taken into account.
+          const csrfToken = getCookie('csrfToken');
+          return Auth.postIdTokenToSessionLogin(API['log-in'],
```

Comment:
ideally you should only do that if the idToken or csrfToken changed from what you had before.